platform/representation/src/gestures/DropGesture.js

```diff
@@ -40,8 +42,9 @@ define(
          * @param {DomainObject} domainObject the domain object whose
          *        composition should be modified as a result of the drop.
          */
-        function DropGesture(dndService, $q, element, domainObject) {
+        function DropGesture(dndService, $q, navigationService, objectService, instantiate, typeService, element, domainObject) {
```

Comment:
This is where most of the 'magic' happens in new edit mode. On drop, the navigated domain object is replaced with an editable version of it.